### ISSUE TITLE
Watch on `events` resource doesn't read updates after an hour or so

### ISSUE THREAD

**Issue by mohitevishal** [splunk/fluent-plugin-kubernetes-objects#5]
Hey,

Has anyone trying to use the watch on events resource encounter an issue where-in the plugin doesn't read events after an hour or so? Below is my config:
```
<source>
  @type kubernetes_objects
  tag "event.*"
  insecure_ssl true
  @log_level trace
  @id in_kubernetes_objects_logs
  <storage>
    @type local
    persistent true
    path /var/log/kubeevents.pos
  </storage>
  <watch>
    resource_name "events"
  </watch>
</source>
```

It reads all the events fine, but doesn't maintain the watch past 1 hour. No errors in logs, it just stops reading new events. Kubernetes by default keeps events only for an hour and removes them one hour after the last occurrence. Well since its a watch this should not matter, but just thought of bringing that up here to see if it has any co-relation with the 1 hour window I get above.

Any pointers to debug this would be greatly appreciated.

Thanks in advance.

Vishal

**Comment by mohitevishal** [comment]
Hey  @Gimi,

Do you have advise on above?

**Comment by dtregonning** [comment]
@mohitevishal Thanks for raising the issue. We think this is related to the default time that the api-server retains events. We believe its directly related to this
https://github.com/kubernetes/kubernetes/issues/52521
Work around suggests setting modifying  below flag
`--event-ttl duration Amount of time to retain events. (default 1h0m0s)`

Reference:
https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/

Let us know how you go modifying and reopen if still an issue

**Comment by mohitevishal** [comment]
@dtregonning thanks for taking a look.
The default `--event-ttl` option simply means, kubernetes will discard the events older than 1 hr. But it will still keep the new events which gets generated.
In the plugin's case, if it's watching a resource say "events", it should continue to get newer events and emit on each new event.

If I am relaying all events to some external source via this plugin, I could even set that `--event-ttl` option to be shorter than 1 hr w/o worrying about losing events.

**Comment by dtregonning** [comment]
Fixed in #27 - however may be issue with volume of events being ingested. #9 to track
